app/javascript/portal/portalThemeHelper.js

```diff
@@ -0,0 +1,146 @@
+import Cookies from 'js-cookie';
+import { adjustColorForContrast } from '../shared/helpers/colorHelper.js';
+
+const SYSTEM_COOKIE = 'system_theme';
+const SELECTED_COOKIE = 'selected_theme';
+
+export const setPortalHoverColor = theme => {
+  if (window.portalConfig.isPlainLayoutEnabled === 'true') return;
+  const portalColor = window.portalConfig.portalColor;
+  const bgColor = theme === 'dark' ? '#151718' : 'white';
+  const hoverColor = adjustColorForContrast(portalColor, bgColor);
+
+  // Set hover color for border and text dynamically
+  document.documentElement.style.setProperty(
+    '--dynamic-hover-color',
+    hoverColor
+  );
+};
+
+export const setPortalClass = theme => {
+  const portalDiv = document.getElementById('portal');
+  if (!portalDiv) return;
+  portalDiv.classList.remove('light', 'dark', 'system');
+  portalDiv.classList.add(theme);
+};
+
+export const updateThemeStyles = theme => {
+  setPortalClass(theme);
+  setPortalHoverColor(theme);
+};
+
+export const toggleAppearanceDropdown = () => {
+  const dropdown = document.getElementById('appearance-dropdown');
+  if (!dropdown) return;
+  const isCurrentlyHidden = dropdown.style.display === 'none';
+  // Toggle the dropdown
+  dropdown.style.display = isCurrentlyHidden ? 'flex' : 'none';
+};
+
+export const updateThemeSelectionInHeader = theme => {
+  // This function is to update the theme selection in the header in real time
+  const buttonContainer = document.getElementById('toggle-appearance');
+  if (!buttonContainer) return;
+  const allElementInButton = buttonContainer.querySelectorAll('.theme-button');
+  const activeThemeElement = document.querySelector(
```

Comment:
See if we can use a data property instead. Also instead of searching the entire document, we can search the container alone